src/stores/windowStore.ts

```diff
@@ -0,0 +1,358 @@
+import create from 'zustand';
+import { Vector2, Vector3 } from 'three';
+import { calculateTilePositions } from '../Utils/tileModes';
+import { Camera } from '@react-three/fiber';
+
+export interface WindowInf {
+  id: string;
+  title: string;
+  subtitle?: string;
+  icon?: string;
+  position: Vector3;
+  content: React.ReactNode;
+  width: number;
+  height: number;
+  isMinimized: boolean;
+  isFullscreen: boolean;
+  scale: Vector3;
+  isFocused?: boolean;
+  isClosable?: boolean;
+  isResizable?: boolean;
+  isMovable?: boolean;
+  onClose?: () => void;
+  onMinimize?: () => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  onClick?: () => void;
+  onHover?: () => void;
+  onHoverEnd?: () => void;
+  disableTitleBar?: boolean;
+  disableIcon?: boolean;
+  disableActionBtns?: boolean;
+  disableBackground?: boolean;
+  disableAdjustSize?: boolean;
+  opacity?: number; // Added opacity option
+}
+
+interface WindowStore {
+  windows: Record<string, WindowInf>;
+  addWindow: (
+    window: Omit<
+      WindowInf,
+      'disableTitleBar' | 'disableIcon' | 'disableActionBtns'
+    > & {
+      disableTitleBar?: boolean;
+      disableIcon?: boolean;
+      disableActionBtns?: boolean;
+    }
+  ) => void;
+  removeWindow: (id: string) => void;
+  updateWindow: (id: string, updates: Partial<WindowInf>) => void;
+  setPosition: (id: string, position: Vector3) => void;
+  setScale: (id: string, scale: Vector3) => void;
+  minimize: (id: string) => void;
+  maximize: (id: string) => void;
+  focus: (id: string) => void;
+  unfocus: (id: string) => void;
+  close: (id: string) => void;
+  tileWindows: (
+    mode: 'grid' | 'around' | 'cockpit',
+    adjustScale?: boolean
+  ) => void;
+  resetWindowPositions: () => void;
+  originalPositions: Record<string, Vector3>;
+  originalScales: Record<string, Vector3>;
+  updateWindowSize: (id: string, size: Vector2) => void;
+  recalculateTilePositions: () => void;
+  currentTileMode: 'grid' | 'around' | 'cockpit' | null;
+  camera: Camera | null;
+  setCamera: (camera: Camera) => void;
+  getPointInFrontOfCamera: (distance: number) => Vector3;
+}
```

Comment:
it seems like some things are internal and other things are public API
at least in the definition i would seperate both
whereever possible do directly expose internal things to the user

Reply:
hmm, you may have a point here, and may want a point set as the look at point or no point at all. Need to consider this.